### ISSUE TITLE
Accordion.Toggle doesn't work inside a NavBar or a Nav

### ISSUE THREAD

**Issue by CsarMan** [react-bootstrap/react-bootstrap#5417]
As the title says, clicking an accordion toggle inside a NavBar doesn't toggle the accordion.

#### Steps to reproduce the behaviour:

1. Click on "AccordionToggle" inside NavBar

#### Reproducible Example

```Javascript
class Header extends React.Component {
  render() {
    return (
      <>
        <Accordion>
          <Accordion.Toggle as={Nav.Link} eventKey="1">
            accordionToogle
          </Accordion.Toggle>
          <Navbar bg="light" expand={true}>
            <Container fluid={"xl"}>
                <Nav className="mr-auto">
                  <Accordion.Toggle as={Nav.Link} eventKey="1">
                    accordionToogle
                  </Accordion.Toggle>
                </Nav>
            </Container>
          </Navbar>
          <Accordion.Collapse eventKey="1">
            <Nav className="bg-light">
              {[1, 2, 3, 4, 5, 6, 7].map((carr) => (
                <Nav.Link key={carr} href={"#"}>
                  {carr}
                </Nav.Link>
              ))}
            </Nav>
          </Accordion.Collapse>
        </Accordion>
      </>
    );
  }
}

```

Note: The Accordion Toggle outside the NavBar it's there just to show that it works outside 

 [CodeSandbox](https://codesandbox.io/s/accordiontoggle-bug-fnjc4).

#### Environment

- Operating System: macOS 15
- Browser, Version Safari 13, Chrome 85
- React-Bootstrap Version 1.3.0

**Comment by kyletsang** [comment]
The Navbar's `SelectableContext` is intercepting the toggle's click here.  Upstream bootstrap docs specifically calls for the usage of cards when creating an accordion so I'm not sure if RB should be allowing different types of header composition like this.

@taion, @jquense thoughts?

**Comment by jquense** [comment]
Yeah, i'm not sure. I think ideally we'd support it but depends on how complex and whatever is breaks. Maybe my comment https://github.com/react-bootstrap/react-bootstrap/pull/5201#issuecomment-667128400 is relevant ?

**Comment by CsarMan** [comment]
Does it have to be limited to cards only? that is kinda very limited... Anyways I somehow thought `<Accordion>` was the `.collapse` alternative here and now I realise that isn't the case. Just found `<Collapse>` and that was just what I needed, but it was just by coincidence. Searching it did not help

![Search](https://user-images.githubusercontent.com/20160120/92299631-f7e28c00-ef21-11ea-97c1-ee02e29d1884.png)
